### PR TITLE
Unit tests for meta sync receive

### PR DESCRIPTION
**Pull request by usmanong**
@VladimirMarkelov @alex-aizman @liangdrew @sasanap 

moving on to meta sync's receiver side, this tests extract smap and lbmab
thanks

### CODE REVIEW

File: dfc/metasync_test.go [hunk 1]
```diff
@@ -669,3 +669,257 @@ func TestMetaSyncMembership(t *testing.T) {
...
+		if !reflect.DeepEqual(a, b) {
+			t.Fatal("SMap mismatch", a, b)
+		}
+	}
+
+	// matchPrevSMap matches two smap partialliy
+	// previous version of the smap is only partially extracted from a sync call
+	matchPrevSMap := func(a, b *Smap) {
+		if a.Version != b.Version {
+			t.Fatal("Previous smap's ersion mismatch", a.Version, b.Version)
```
Comment (liangdrew) [inline dfc/metasync_test.go]: `ersion` -> `version`

File: dfc/metasync_test.go [hunk 2]
```diff
@@ -669,3 +669,257 @@ func TestMetaSyncMembership(t *testing.T) {
...
+			_, ok := b.Tmap[v.DaemonID]
+			if !ok {
+				t.Fatal("Missing target", v.DaemonID)
+			}
+		}
+	}
+
+	nilLBMap := func(l *lbmap) {
+		if l != nil {
+			t.Fatal("Expecting nil lsmap", l)
```
Comment (liangdrew) [inline dfc/metasync_test.go]: `lsmap` -> `lbmap`

File: dfc/metasync_test.go [hunk 3]
```diff
@@ -669,3 +669,257 @@ func TestMetaSyncMembership(t *testing.T) {
...
+		}
+	}
+
+	matchSMap := func(a, b *Smap) {
+		if !reflect.DeepEqual(a, b) {
+			t.Fatal("SMap mismatch", a, b)
+		}
+	}
+
+	// matchPrevSMap matches two smap partialliy
```
Comment (liangdrew) [inline dfc/metasync_test.go]: `partialliy` -> `partially`

File: dfc/metasync_test.go [hunk 4]
```diff
@@ -669,3 +669,257 @@ func TestMetaSyncMembership(t *testing.T) {
...
+	prevSMap = primary.smap.cloneU()
+
+	s = httptest.NewServer(http.HandlerFunc(fTarget))
+	defer s.Close()
+	primary.smap.add(&daemonInfo{DaemonID: "target2", DirectURL: s.URL})
+	target2 := proxyrunner{}
+	target2.smap = &Smap{Tmap: make(map[string]*daemonInfo), Pmap: make(map[string]*proxyInfo)}
+	target1.lbmap = newLBMap()
+
+	c = func(n, o *Smap, m *ActionMsg, e string) {
```
Comment (VladimirMarkelov) [inline dfc/metasync_test.go]: What is the difference between this `c` and `c` at line 825?

Comment (usmanong) [reply dfc/metasync_test.go]: haha, nice catch, use to be different, now they are the same, thank you

**Comment by usmanong** [comment]
updated diff

File: dfc/metasync_test.go (hunk 1 above)
Comment (usmanong) [reply dfc/metasync_test.go]: all fixed, thank you

File: dfc/metasync_test.go [hunk 5]
```diff
@@ -669,3 +669,258 @@ func TestMetaSyncMembership(t *testing.T) {
...
+	noErr(errStr)
+	nilLBMap(lb)
+	emptyActionMsg(actMsg)
+
+	// older version
+	proxy.lbmap.Version++
+	_, _, errStr = proxy.extractlbmap(payload)
+	hasErr(errStr)
+
+	syncer.stop(nil)
```
Comment (liangdrew) [inline dfc/metasync_test.go]: If an early return occurs, the metasyncer won't stop, right? Is this an issue?

**Pull request merged**